### DISCUSSION TITLE
to track a directory for LFS.

### DISCUSSION THREAD

**Discussion by JLuisAlcaraz** [git-lfs/git-lfs#5271] [Q&A]
Hi all,
I want to track an entire directory to lfs.
I want anything, inside that directory will be tracked by lfs, including nested directories.

I use "git lfs track myDir/*", but this creates a serie of lines, inside .gitattributes, one by nested directory.
The problem is when I replace all content, inside "myDir", including new directories. In that case, git won't track the news directories, and in order to do that, I need to modify .gitattributes.

Is there a way to lfs track the content of a dir, no matter what it is (files, directories, etc).

Thks in advance!

**Comment by bk2204** [comment]
Hey,

That's because you didn't quote the `myDir/*` and so the shell expanded that to all of the contents of `myDir`.  What you want to do instead is track `myDir/**`, which will track everything under `myDir` recursively.  Note that you need to write this on the command line as `git lfs track 'myDir/**'` (or, on Windows, `git lfs track "myDir/**"`), using proper quoting so that the shell doesn't expand it.

**Reply by JLuisAlcaraz** [reply]
Thank you very much!

**Reply by s3xysteak** [reply]
I don't get it. I have the same problem that I want to track a folder but I do not care what it contains. Like a `public` folder contains some libs or video and it may be add more files everyday. Should I just use `git lfs track public` or `git lfs track public/**` ?

**Reply by s3xysteak** [reply]
I found use `git lfs track myDir/**/*` will track everything in it. But some team member may forgot to use the command while adding a new file, so I think could I just track this folder, just like using `myDir` in .gitignore?

**Reply by bk2204** [reply]
The patterns are different in `.gitignore` and `.gitattributes`.  The former allows listing a directory to apply to everything in it and the latter does not.  Therefore, you need to use `public/**`.  `git help gitattributes` mentions this fact.

If you're running on the command line, you need to appropriately quote the value.  Thus, on Linux, macOS, or other Unix systems, you need to use something like `git lfs track 'public/**'` and on Windows you should use double quotes instead of single quotes there instead.  If you don't do that, the shell will expand the paths and you won't get the correct behaviour.

**Reply by s3xysteak** [reply]
Is `.gitattributes` unable to be used on a folder? Can it only be used on a file?

**Reply by bk2204** [reply]
A pattern in `.gitattributes` which matches only a folder (and doesn't contain something like `/**`) matches nothing.  That's what's documented in the `gitattributes` manual page.  You have to match some file or collection of files instead.
